mergify_engine/engine/__init__.py

```diff
@@ -79,6 +79,25 @@ async def _check_configuration_changes(
         # As the PR is closed, we don't care about the config change detector.
         return False
 
+    # NOTE(sileht): This heuristic works only if _ensure_summary_on_head_sha()
+    # is called after _check_configuration_changes().
+    # If we don't have a summary yet, it means the pull request has just been
+    # open or synchronize or we never see it.
+    summary = await ctxt.get_engine_check_run(constants.SUMMARY_NAME)
+    print(summary)
+    if (
+        summary
+        and summary["output"]["title"]
+        != constants.CONFIGURATION_MUTIPLE_FOUND_SUMMARY_TITLE
+    ):
+        configuration_changed_check = await ctxt.get_engine_check_run(
+            constants.CONFIGURATION_CHANGED_CHECK_NAME
+        )
+        configuration_deleted_check = await ctxt.get_engine_check_run(
+            constants.CONFIGURATION_DELETED_CHECK_NAME
+        )
+        return bool(configuration_changed_check) or bool(configuration_deleted_check)
```

Comment:
```suggestion
        configuration_changed_check = await ctxt.get_engine_check_run(
            constants.CONFIGURATION_CHANGED_CHECK_NAME
        )
        if configuration_changed_check:
            return True
        configuration_deleted_check = await ctxt.get_engine_check_run(
            constants.CONFIGURATION_DELETED_CHECK_NAME
        )
        return bool(configuration_deleted_check)
```
avoid running 2 awaits if not needed?

Reply:
all checks are always retrieved.